### PR TITLE
The forcing is variable, and user defined at the cases level

### PR DESCRIPTION
**Pull request by flampouris**
## Description
With this PR, the user can define at the case's yaml file, what forcing to use.

Secondary: Fixed a couple of typos at the README.

### Issue(s) addressed
Simplifies the testing with the different forcing

## Testing
@JessicaMeixner-NOAA see the updates at the prep_forecast.sh and the forecast case
@jkbk2004 run the 3dvar
@JianKuang-NOAA check if the updates at the workflow are at the appropriate files or could be somewhere else

## Dependencies
No dependencies

### CODE REVIEW

File: scripts/prep_forecast.sh [hunk 1]
```diff
@@ -66,12 +66,24 @@ ATMPETS=${ATMPETS:-"72"}   #Number of ATM Pets
...
-
+DATM_FILENAME_BASE=${DATM_FILENAME_BASE:-"${FORCING_SRC,,}."} #The prefix of the forcing files for the DATM
+
+#nfhout number of hours between DATM inputs 6 for cfsr 3 for gefs
+#dimension of DATM input files, lon
+#dimension of DATM input files, lat
+
+if [ "${FORCING_SRC,,}" = "cfsr" ]; then
+   NFHOUT=${NFHOUT:-6}
+   IATM=${IATM:-1760}
```
Comment (JessicaMeixner-NOAA) [inline scripts/prep_forecast.sh]: This is correct IATM, JATM for 20111001:

ncdump -h /scratch2/NCEPDEV/marineda/godas_input/DATM_INPUT/CFSR/201110/cfsr.2011100206.nc
netcdf cfsr.2011100206 {
dimensions:
	lon = 1760 ;
	lat = 880 ;


However, for another date: 
ncdump -h /scratch2/NCEPDEV/marineda/godas_input/DATM_INPUT/CFSR/199304/cfsr.1993041012.nc
netcdf cfsr.1993041012 {
dimensions:
	lon = 1152 ;
	lat = 576 ;
	time = UNLIMITED ; // (1 currently)

So apparently the dimensions of these files change in time.  Not sure what date that change happens at.

File: scripts/prep_forecast.sh [hunk 2]
```diff
@@ -66,12 +66,24 @@ ATMPETS=${ATMPETS:-"72"}   #Number of ATM Pets
...
+   JATM=${JATM:-880}
+elif [ "${FORCING_SRC,,}" = "gefs" ]; then
+   NFHOUT=${NFHOUT:-3}
+   IATM=${IATM:-1536}
+   JATM=${JATM:-768}
```
Comment (JessicaMeixner-NOAA) [inline scripts/prep_forecast.sh]: Defaults for GEFS are correct and file size seems to be consistent: 

ncdump -h /scratch2/NCEPDEV/marineda/godas_input/DATM_INPUT/GEFS/200107/gefs.2001072000.nc
netcdf gefs.2001072000 {
dimensions:
	lon = 1536 ;
	lat = 768 ;
	time = UNLIMITED ; // (1 currently)

ncdump -h /scratch2/NCEPDEV/marineda/godas_input/DATM_INPUT/GEFS/201110/gefs.2011100100.nc 
netcdf gefs.2011100100 {
dimensions:
	lon = 1536 ;
	lat = 768 ;

**Review by JessicaMeixner-NOAA — APPROVED** [review]
I did not run this test, however the changes themselves look good!

File: scripts/prep_forecast.sh (hunk 1 above)
Comment (JessicaMeixner-NOAA) [reply scripts/prep_forecast.sh]: Looking at sizes of files, you would need 
   IATM=${IATM:-1760}
    JATM=${JATM:-880}
for the 2011 cases, 
and 
   IATM=${IATM:-1152}
    JATM=${JATM:-576}
for all other dates (currently converted and on hera) for CFSR.

**Comment by flampouris** [comment]
@JessicaMeixner-NOAA 

Please review the code again, now the IATM and JATM are variables imported from the forcing files.

**Review by JessicaMeixner-NOAA — APPROVED** [review]
Looks good, I did not test though, just fyi

**Comment by jkbk2004** [comment]
@JessicaMeixner-NOAA @flampouris I am testing the branch now.

**Review by jkbk2004 — APPROVED** [review]
fcst and 3dvar run ok!

**Comment by flampouris** [comment]
Thank you all!

**Pull request merged**